lib/libfrr.c

```diff
@@ -846,7 +846,8 @@ void frr_fini(void)
 	/* memory_init -> nothing needed */
 	vty_terminate();
 	cmd_terminate();
-	zprivs_terminate(di->privs);
+	if (di->privs->user)
```

Comment:
this check needs to be moved to zprivs_terminate (because the "other" check for this on startup is in zprivs_init)